spi/src/main/java/org/opensearch/jobscheduler/spi/LockModel.java

```diff
@@ -89,36 +171,119 @@ public LockModel(final LockModel copyLock, final boolean released) {
+        if (copyLock.isResourceLock) {
+            this.lockId = copyLock.lockId;
+            this.resource = copyLock.resource;
+            this.resourceType = copyLock.resourceType;
+            this.isResourceLock = true;
+        }
+        else {
+            this.lockId = copyLock.jobIndexName + LOCK_ID_DELIMITER + copyLock.jobId;
+            this.resource = null;
+            this. resourceType = null;
+            this.isResourceLock = false;
+        }
+        this.jobIndexName = copyLock.jobIndexName;
+        this.jobId = copyLock.jobId;
+        this.seqNo = copyLock.seqNo;
+        this.primaryTerm = copyLock.primaryTerm;
+        this.lockTime = updateLockTime;
+        this.lockDurationSeconds = lockDurationSeconds;
+        this.released = released;
     }
 
+    /**
+     * Use this constructor to build a job-scheduler lock.
+     * @param jobIndexName          index of job acquiring the lock.
+     * @param jobId                 id of job acquiring the lock.
+     * @param lockTime              time at which the lock is acquired.
+     * @param lockDurationSeconds   duration for which the lock is valid.
+     * @param released              boolean flag to indicate if the lock is released.
+     */
     public LockModel(String jobIndexName, String jobId, Instant lockTime, long lockDurationSeconds, boolean released) {
         this(jobIndexName, jobId, lockTime, lockDurationSeconds, released,
-            SequenceNumbers.UNASSIGNED_SEQ_NO, SequenceNumbers.UNASSIGNED_PRIMARY_TERM);
+                SequenceNumbers.UNASSIGNED_SEQ_NO, SequenceNumbers.UNASSIGNED_PRIMARY_TERM);
     }
 
+    /**
+     * Fully parametrized constructor for building a job-scheduler lock
+     * @param jobIndexName          index of job acquiring the lock.
+     * @param jobId                 id of job acquiring the lock.
+     * @param lockTime              time at which the lock is acquired.
+     * @param lockDurationSeconds   duration for which the lock is valid.
+     * @param released              boolean flag to indicate if the lock is released.
+     * @param seqNo                 sequence number from OpenSearch document.
+     * @param primaryTerm           primary term from OpenSearch document.
+     */
     public LockModel(String jobIndexName, String jobId, Instant lockTime,
                      long lockDurationSeconds, boolean released, long seqNo, long primaryTerm) {
-        this.lockId = jobIndexName + LOCK_ID_DELIMITR + jobId;
+        System.out.println("Job id passed in "+ jobId);
+        this.lockId = jobIndexName + LOCK_ID_DELIMITER + jobId;
+        this.jobIndexName = jobIndexName;
+        this.jobId = jobId;
+        this.lockTime = lockTime;
+        this.lockDurationSeconds = lockDurationSeconds;
+        this.released = released;
+        this.seqNo = seqNo;
+        this.primaryTerm = primaryTerm;
+        this.resource = null;
+        this.resourceType = null;
+        this.isResourceLock = false;
```

Comment:
Can you call the fully parameterized constructor here instead in line 83 to prevent having multiple places with setting logic?